client/ayon_core/pipeline/traits/representation.py

```diff
@@ -0,0 +1,710 @@
+
+        """
+        return bool(self._data)
+
+    def contains_trait(self, trait: Type[T]) -> bool:
+        """Check if the trait exists in the Representation.
+
+        Args:
+            trait (TraitBase): Trait class.
+
+        Returns:
+            bool: True if the trait exists, False otherwise.
+
+        """
+        return bool(self._data.get(str(trait.id)))
+
+    def contains_trait_by_id(self, trait_id: str) -> bool:
+        """Check if the trait exists using trait id.
+
+        Args:
+            trait_id (str): Trait ID.
+
+        Returns:
+            bool: True if the trait exists, False otherwise.
+
+        """
+        return  bool(self._data.get(trait_id))
+
+    def contains_traits(self, traits: list[Type[T]]) -> bool:
+        """Check if the traits exist.
+
+        Args:
+            traits (list[TraitBase], optional): List of trait classes.
+
+        Returns:
+            bool: True if all traits exist, False otherwise.
+
+        """
+        return all(self.contains_trait(trait=trait) for trait in traits)
+
+    def contains_traits_by_id(self, trait_ids: list[str]) -> bool:
+        """Check if the traits exist by id.
+
+        If no trait IDs or traits are provided, it will check if the
+        representation has any traits.
+
+        Args:
+            trait_ids (list[str]): List of trait IDs.
+
+        Returns:
+            bool: True if all traits exist, False otherwise.
+
+        """
+        return all(
+            self.contains_trait_by_id(trait_id) for trait_id in trait_ids
+        )
+
+    def get_trait(self, trait: Type[T]) -> T:
+        """Get a trait from the representation.
+
+        Args:
+            trait (TraitBase, optional): Trait class.
+
+        Returns:
+            TraitBase: Trait instance.
+
+        Raises:
+            MissingTraitError: If the trait is not found.
+
+        """
+        try:
+            return self._data[str(trait.id)]
+        except KeyError as e:
+            msg = f"Trait with ID {trait.id} not found."
+            raise MissingTraitError(msg) from e
+
+    def get_trait_by_id(self, trait_id: str) -> T:
+        # sourcery skip: use-named-expression
+        """Get a trait from the representation by id.
+
+        Args:
+            trait_id (str): Trait ID.
+
+        Returns:
+            TraitBase: Trait instance.
+
+        Raises:
+            MissingTraitError: If the trait is not found.
+
+        """
+        version = _get_version_from_id(trait_id)
+        if version:
+            try:
+                return self._data[trait_id]
+            except KeyError as e:
+                msg = f"Trait with ID {trait_id} not found."
+                raise MissingTraitError(msg) from e
+
+        result = next(
+            (
+                self._data.get(trait_id)
+                for trait_id in self._data
+                if trait_id.startswith(trait_id)
+            ),
+            None,
+        )
+        if not result:
```

Comment:
```suggestion
        if result is None:
```

Reply:
this cannot hit valid boolean value, but yeah...